### DISCUSSION TITLE
WG easy in docker compose with other services

### DISCUSSION THREAD

**Discussion by roycw** [wg-easy/wg-easy#1190] [Q&A]
Hello, i have a docker compose per the example with other services that are also sonnected to the network so they can get access to the wg network. when i try to expose other ports on the other continers i get an error from the dokcer deamon stating that i can not do that. ias that possible to add conteiner that are part of the network and also expose their own ports...

**Comment by eandersons** [comment]
I do not see any reason why you could not add other containers to the same network as WireGuard Easy container and expose their ports unless you are trying to publish port that is already used on the host.

One of the ways you could check if port is available: `netstat -ltn | grep <port in question>` (there will be no output if the port is not taken).

It would help if you provided the exact error and ideally also excerpts from your Compose file (especially `ports` blocks).

**Comment by roycw** [comment]
hee is an example of my dc
~~~
version: '3.9'

volumes:
  etc_wireguard:

services:

  wg:
    image: ghcr.io/wg-easy/wg-easy
    environment:
      - WG_HOST=3.148.107.36
    volumes:
      - etc_wireguard:/etc/wireguard
    ports:
      - "51820:51820/udp"
      - "51821:51821/tcp"
    restart: unless-stopped
    cap_add:
      - NET_ADMIN
      - SYS_MODULE
    sysctls:
      - net.ipv4.ip_forward=1
      - net.ipv4.conf.all.src_valid_mark=1

  go2rtc:
    image: alexxit/go2rtc:latest
    restart: unless-stopped  # autorestart on fail or config change from WebUI
    environment:
      - TZ=Atlantic/Bermuda  # timezone in logs
    volumes:
      - /app:/config
    network_mode: "service:wg"
    depends_on:
      - wg

  agent:
    image: myregistry
    network_mode: "service:wg"
    restart: always
    command: -c /app/agent.toml
    volumes:
      - /app/agent.toml:/app/agent.toml
    depends_on:
      - wg

~~~

i think the issue is in the network_mode
If i remove that then i can not access wg0 from other containers.....

**Comment by roycw** [comment]
hi [eandersons](https://github.com/eandersons) did u find the issue in the docker compose ?

**Reply by eandersons** [reply]
Hi,

No, I did not find any obvious issue.

On the other hand it is not clear what exactly are you trying to achieve, so probably there is another way(s) to get the desired outcome.

> when i try to expose other ports on the other continers i get an error from the dokcer deamon stating that i can not do that.

You most probably should publish the needed ports on the WireGuard Easy container since the other services are using its network.
